service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -157,6 +161,13 @@ public ResponseEntity<RunSetListResponse> getRunSets(UUID methodId, Integer page
       filteredRunSet = runSetDao.getRunSets(pageSize, false);
     }
 
+    if (Boolean.FALSE.equals(showAllUsers)) {
+      filteredRunSet =
+          filteredRunSet.stream()
+              .filter(runSet -> runSet.userId().equals(userInfo.getUserSubjectId()))
```

Comment:
TOL: I wonder if it would be better to filter based on User ID at the database layer rather than here. Apart from it being little faster it would also mean sending less data from database to application layer.